src/libraries/System.Linq/tests/AggregateByTests.cs

```diff
@@ -0,0 +1,211 @@
+                expected: Enumerable.Range(0, 5).ToDictionary(x => x, x => 30 + 4 * x));
+
+            yield return WrapArgs(
+                source: Enumerable.Repeat(5, 20),
+                keySelector: x => x,
+                seedSelector: x => 0,
+                func: (x, y) => x + y,
+                comparer: null,
+                expected: Enumerable.Repeat(5, 1).ToDictionary(x => x, x => 100));
+
+            yield return WrapArgs(
+                source: new string[] { "Bob", "bob", "tim", "Bob", "Tim" },
+                keySelector: x => x,
+                seedSelector: x => string.Empty,
+                func: (x, y) => x + y,
+                null,
+                expected: new Dictionary<string, string>
+                {
+                    { "Bob", "BobBob" },
+                    { "bob", "bob" },
+                    { "tim", "tim" },
+                    { "Tim", "Tim" },
+                });
+
+            yield return WrapArgs(
+                source: new string[] { "Bob", "bob", "tim", "Bob", "Tim" },
+                keySelector: x => x,
+                seedSelector: x => string.Empty,
+                func: (x, y) => x + y,
+                StringComparer.OrdinalIgnoreCase,
+                expected: new Dictionary<string, string>
+                {
+                    { "Bob", "BobbobBob" },
+                    { "tim", "timTim" }
+                });
+
+            yield return WrapArgs(
+                source: new (string Name, int Age)[] { ("Tom", 20), ("Dick", 30), ("Harry", 40) },
+                keySelector: x => x.Age,
+                seedSelector: x => $"I am {x} and my name is ",
+                func: (x, y) => x + y.Name,
+                comparer: null,
+                expected: new Dictionary<int, string>
+                {
+                    { 20, "I am 20 and my name is Tom" },
+                    { 30, "I am 30 and my name is Dick" },
+                    { 40, "I am 40 and my name is Harry" }
+                });
+
+            yield return WrapArgs(
+                source: new (string Name, int Age)[] { ("Tom", 20), ("Dick", 20), ("Harry", 40) },
+                keySelector: x => x.Age,
+                seedSelector: x => $"I am {x} and my name is ",
+                func: (x, y) => $"{x} or {y.Name}",
+                comparer: null,
+                expected: new Dictionary<int, string>
+                {
+                    { 20, "I am 20 and my name is Tom or Dick" },
+                    { 40, "I am 40 and my name is Harry" }
+                });
+
+            yield return WrapArgs(
+                source: new (string Name, int Age)[] { ("Bob", 20), ("bob", 20), ("Harry", 20) },
+                keySelector: x => x.Name,
+                seedSelector: x => 0,
+                func: (x, y) => x + y.Age,
+                comparer: null,
+                expected: new string[] { "Bob", "bob", "Harry" }.ToDictionary(x => x, x => 20));
+
+            yield return WrapArgs(
```

Comment:
For completeness, could you also add a few test cases illustrating:

1. implementation of `GroupBy` in terms of `AggregateBy`.
2. implementation of `LongCountBy` in terms of `AggregateBye` and
3. score aggregation

Sample data can be found in the "API usage" section of the proposal https://github.com/dotnet/runtime/issues/91533

Reply:
Tests added.